cli/service/service.go

```diff
@@ -135,12 +135,22 @@ func Log(serviceName string, tailing bool) {
 					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Deployment unmarshalling error: %v", err))
 				}
 
-				for _, service := range deployment.Services {
-					var err error
-					if nonDefaultLogDriverUsed, err = cliutils.ChekServiceLogPossibility(service.LogDriver); err != nil {
-						cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("Service logs are unavailable: %v", err))
+				containerFound := false
+				if len(deployment.Services) > 1 {
+					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Service definition %v consists of more than one container. Please specify the container name using the -c flag.", serviceName))
+				}
+				for deployedServiceName, service := range deployment.Services {
+					if containerName == deployedServiceName || (containerName == "" && len(deployment.Services) == 1) {
+						var err error
+						if nonDefaultLogDriverUsed, err = cliutils.ChekServiceLogPossibility(service.LogDriver); err != nil {
+							cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, msgPrinter.Sprintf("Service logs are unavailable: %v", err))
+						}
+						containerFound = true
+						break
 					}
-					break
+				}
+				if !containerFound {
+					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Container %v is not running as part of service %v.", containerName, serviceName))
```

Comment:
It is going to seem odd to the user when this message (or the one at line 140) comes out AFTER the message on line 109. Can we move the msgs at 109 down to after 159?